samcli/lib/package/ecr_uploader.py

```diff
@@ -83,6 +90,40 @@ def upload(self, image, resource_name):
 
         return f"{repository}:{_tag}"
 
+    def delete_artifact(self, image_uri: str, resource_id: str, property_name: str):
+        """
+        Delete the given ECR image by extracting the repository and image_tag from
+        image_uri
+
+        :param image_uri: image_uri of the image to be deleted
+        :param resource_id: id of the resource for which the image is deleted
+        :param property_name: provided property_name for the resource
+        """
+        try:
+            repo_image_tag = image_uri.split("/")[1].split(":")
+            repository = repo_image_tag[0]
+            image_tag = repo_image_tag[1]
+            resp = self.ecr_client.batch_delete_image(
+                repositoryName=repository,
+                imageIds=[
+                    {"imageTag": image_tag},
+                ],
+            )
+            if resp["failures"]:
+                # Image not found
+                image_details = resp["failures"][0]
+                if image_details["failureCode"] == "ImageNotFound":
+                    LOG.error("ImageNotFound Exception : ")
+                    raise ImageNotFoundError(resource_id, property_name)
```

Comment:
This exception will show this  message `Image not found for {property_name} parameter of {resource_id} resource. \n` to the customer, I prefer something like `Could delete Image {} used for {} resource because it is not exist`